src/main/java/seedu/address/logic/commands/ImportCommand.java

```diff
@@ -0,0 +1,164 @@
+    private final String filePath;
+
+    /**
+     * Creates an ImportCommand to import contacts from the specified CSV file path.
+     *
+     * @param filePath The path of the CSV file to import contacts from.
+     */
+    public ImportCommand(String filePath) {
+        requireNonNull(filePath);
+        this.filePath = filePath;
+    }
+
+    /**
+     * Executes the ImportCommand, reading contacts from the specified CSV file
+     * and adding them to the address book.
+     *
+     * @param model The model in which contacts are to be added.
+     * @return A CommandResult indicating success or failure of the import operation.
+     * @throws CommandException If an error occurs during file reading or processing.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        int successCount = 0;
+
+        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
+            String line = br.readLine(); // Skip the header row
+
+            while ((line = br.readLine()) != null) {
+                String[] values = line.split(","); // Split CSV by commas
+
+                if (values.length < 6) {
+                    throw new CommandException("Invalid CSV format");
+                }
+
+                Name name = new Name(values[0].trim());
+                String category = values[1].trim();
+                Phone phone = new Phone(values[3].trim());
+                Email email = new Email(values[4].trim());
+                Address address = new Address(values[5].trim());
+                Set<Tag> tagSet = parseTags(values.length > 6 ? values[6].trim() : "");
+
+                switch (category.toLowerCase()) {
+                case "student":
+                    String studentID = values[2].trim();
+                    if (!studentID.isEmpty()) {
+                        Student student = new Student(name, new StudentID(studentID), phone, email, address, tagSet);
+                        if (!model.hasPerson(student)) {
+                            new AddStudentCommand(student).execute(model);
+                            successCount++;
+                        }
+                    } else {
+                        throw new CommandException("Missing Student ID for student category");
+                    }
+                    break;
+
+                case "company":
+                    String industry = values[2].trim();
+                    if (!industry.isEmpty()) {
+                        Company company = new Company(name, new Industry(industry), phone, email, address, tagSet);
+                        if (!model.hasPerson(company)) {
+                            new AddCompanyCommand(company).execute(model);
+                            successCount++;
+                        }
+                    } else {
+                        throw new CommandException("Missing Industry for company category");
```

Comment:
Same here can add a test for this

Reply:
Yes, can maybe add an `assertThrows()` test

Reply:
Thank you for the advice!